### ISSUE TITLE
docs: showcase differentiable rendering, and unify the figure gallery

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#44]
Every one of the fourteen figures on the showcase is a **forward** operation — extract a surface, query a hierarchy, measure a mesh. Nothing on the site demonstrates the property the library leads with in its own overview: that gradients flow back through the extractors into the field.

`examples/differentiable_rendering/main.py` already does exactly that, optimising a raw SDF grid into a target shape under multi-view silhouette and depth loss with `diff_marching_cubes` carrying the gradient. It has never been shown.

Two further problems with the showcase page as it stands:

- **Two card sizes on one page.** The lead comparisons span the full width while the supporting figures flow in three columns, so the page has two reading rhythms and an arbitrary split between "Qualitative results" and "Others".
- The acknowledgement records neither the SuperFlex paper that `conquer3d/primitive/sq.py` cites in its own module docstring, nor the repositories and reference page behind it.

**Comment by KhoiDOO** [comment]
Landed on `docs/diffrender-showcase` in six commits.

## API usage

The figure runs the library's public differentiable path end to end. This is the loop, reduced to what matters — it is the `--method mc` branch of `examples/differentiable_rendering/main.py`:

```python
import torch, kaolin as kal, nvdiffrast.torch as dr
from conquer3d.data_structure import create_voxel_grid
from conquer3d.ops import diff_marching_cubes

grid_vertices, voxels, _ = create_voxel_grid(
    grid_min=[-1.0] * 3, grid_max=[1.0] * 3, res=[128] * 3, device="cuda"
)

# A field of noise. ~10% of corners start inside, so iteration 0 is a foam.
sdf = torch.nn.Parameter((torch.rand_like(grid_vertices[:, 0]) - 0.1).clone().detach())
optimizer = torch.optim.Adam([sdf], lr=0.01)
glctx = dr.RasterizeCudaContext(device="cuda")

for it in range(1000):
    optimizer.zero_grad()
    cameras = render.get_random_camera_batch(8, iter_res=[2048, 2048], device="cuda")
    target = render.render_mesh(glctx, gt_mesh, cameras, [2048, 2048],
                                return_types=["mask", "depth"])

    # (V, 3) float32 with grad, (F, 3) int32. Returns a 4-tuple; the mesh is the first two.
    verts, faces = diff_marching_cubes(grid_vertices, voxels, sdf, iso=0.0)[:2]

    buffers = render.render_mesh(
        glctx, kal.rep.SurfaceMesh(vertices=verts, faces=faces.long()),
        cameras, [2048, 2048], return_types=["mask", "depth"])

    mask_loss = (buffers["mask"] - target["mask"]).abs().mean()
    depth_loss = ((((buffers["depth"] - target["depth"]) * target["mask"]) ** 2)
                  .sum(-1) + 1e-8).sqrt().mean() * 10.0
    (mask_loss + depth_loss).backward()      # gradient reaches sdf through the extractor
    optimizer.step()
```

No loss term touches the surface directly. The only supervision is a rendered silhouette mask and a depth buffer; `diff_marching_cubes` is the whole path from pixels to field.

Rebuild the figure with:

```bash
~/anaconda3/envs/geocutool/bin/python docs/_figures/make_figures.py diffrender
PATH="$HOME/anaconda3/bin:$PATH" python3 docs/generate.py --strict
```

## Advantages demonstrated

- **The extractor is genuinely differentiable, not differentiable in principle.** A field initialised to uniform noise reaches a recognisable statue in 30 iterations and converges in under 200, driven only by image-space loss.
- **Fast enough to be a training step, not a preprocessing stage.** 1,000 iterations — each one a full marching-cubes extraction, two 8-view rasterisations at 2048x2048, and a backward pass — complete in **32 s** on an RTX 4090, peak 6.9 GiB. That is ~32 ms per optimisation step at 128^3.
- **Output stays compact as it converges.** Face count falls from 1,633,759 to 126,960 as the foam resolves into a surface, ending 29% *below* the 98,601-face source only because marching cubes tessellates a smooth field more evenly than the scanned original.

## Trade-offs and known limits

- **The 2048x2048 training resolution is doing real work.** A probe at 512x512 converged visibly worse per iteration (CD 0.327 vs 0.169 at iteration 12), so the silhouette gradient is resolution-hungry. Dropping render resolution is the wrong first economy.
- **Chamfer distance plateaus early while loss keeps falling.** CD is 0.088 at iteration 30 and 0.060 at 1,000 — a 32% improvement — while loss falls 33x over the same span. Mask and depth loss saturate on silhouette agreement long before fine geometry is right. This is the same ceiling reported in #43.
- **Iteration 0 is the memory high-water mark**, not the end: 1.6M faces of noise must be extracted, rasterised and differentiated before the field has any structure. Peak allocation is set by the initialisation, so a coarser `--batch` helps more than a coarser grid.
- **The figure depends on `examples/`.** `docs/` had never imported from `examples/` before. Both ship in the repo, neither ships in the wheel, and `make_figures.py` is run by hand, so no build path is affected — but the import raises rather than degrading if the example moves.

## Measured results

Happy Buddha, 128^3 grid, 8 views at 2048x2048, Adam at lr 0.01 with exponential decay, seeded:

| iteration | faces | loss | Chamfer |
| ---: | ---: | ---: | ---: |
| 0 | 1,633,759 | 1.545 | 0.549 |
| 10 | 399,664 | 0.938 | 0.349 |
| 30 | 173,564 | 0.281 | 0.088 |
| 75 | 185,936 | 0.066 | 0.069 |
| 1000 | 126,960 | 0.008 | 0.060 |
| source | 98,601 | — | reference |

Checkpoints were chosen after measuring ten of them: at the original spacing (0, 50, 200, 500) the last four panels were visually identical, because loss is already 0.012 by iteration 200.

For comparison, the same configuration on the Armadillo reaches loss 0.011 / CD 0.069 — the Buddha converges further despite being the more detailed asset.

## Technical insights worth preserving

- **A module-name collision that fails silently.** `docs/_figures/make_figures.py` does `sys.path.insert(0, HERE)` then `import render as R`, so `sys.modules["render"]` is already the docs renderer. The example's helper is *also* `render.py`. A plain `import render` returns the cached docs module — no error, just a missing `render_mesh`. It is loaded by path under its own name via `importlib.util.spec_from_file_location` instead.
- **Two renderers, deliberately.** `docs/_figures/render.py` states a no-kaolin rule in its own docstring. That rule is kept: kaolin drives only the differentiable training loop, while every panel is drawn by the site's own renderer at a **fixed** camera (`fit_radius=1.0`, no per-mesh refit), so the trajectory reads as changing shape rather than changing size.
- **Camera azimuth is load-bearing for a figure's message.** The first render sat at azimuth 205 — behind the statue, where the robe is broad and featureless and a converged reconstruction is indistinguishable from a rough one. Rotating to azimuth 25 put the face and attendant figures in view, which is what makes the last two panels show detail the optimisation actually had to recover.
- **Deleting one CSS rule rewrote 818 files.** Removing the dead `.fig--lead` block changed `site.css`, and `relink_assets` re-fingerprints `assets/css/site.css?v=` on every generated page, frozen v0.7.x/v0.8.x API archives included. Isolated into its own commit so it does not bury the feature.
- **Uniform cards cost the widest figures.** At a 1440px viewport each card is ~453px, so `fig-sign-modes` (3592x595) renders 68px tall and `fig-resolution` (2000x373) 76px. Everything at 119px and above stays legible. The lightbox binds to `.fig-media` and is unaffected by the merge, so a sliver is still one click from full size; reflowing those two builders to two rows via `cols=` would fix it properly.

**Comment by KhoiDOO** [comment]
Implemented and covered by PR #45.
